Doctor validation fails when CI is still pending after commit

## Problem

During shepherd orchestration for #1901, the doctor phase made a valid commit to fix E2E tests, but shepherd validation declared failure because CI hadn't completed yet:

```
[06:55:41] ⟳ doctor running (3m elapsed)
[06:56:26] [ERROR] Doctor made 1 commit(s) but validation failed. Label state may need manual recovery.
```

The doctor's commit (69e9ff1 - "Fix E2E tests for single-session UI model") was valid and CI eventually passed, but the shepherd exited with error code 4 because validation ran before CI completed.

## Expected Behavior

Shepherd should either:
1. Wait for CI to complete before validating doctor's work
2. Or distinguish between "doctor commit failed CI" vs "CI still pending"
3. Or retry validation after a delay when CI is pending

## Current Behavior

Shepherd runs validation immediately after doctor commits, and if CI checks are pending/running, it declares validation failure and exits.

## Impact

- Requires manual re-run of shepherd to continue
- False positive "validation failed" errors
- Label state warnings that don't reflect actual problems

## Suggested Fix

In the doctor phase validation:
1. Check CI status after doctor commits
2. If CI is pending/running, wait for completion (with timeout)
3. Only declare failure if CI actually fails, not if it's still running

## Context

- Issue: #1901
- PR: #2080
- Doctor commit: 69e9ff1672737c7276958545be514ebd088f80c9

--- a/loom-tools/src/loom_tools/shepherd/phases/doctor.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/doctor.py
@@ -3,17 +3,61 @@
 from __future__ import annotations
 
 import subprocess
+import time
 from dataclasses import dataclass
 from enum import Enum
 from pathlib import Path
 from typing import Any
 
+from loom_tools.common.logging import log_info, log_warning
 from loom_tools.shepherd.context import ShepherdContext
 from loom_tools.shepherd.phases.base import (
     PhaseResult,
     PhaseStatus,
     run_phase_with_retry,
 )
+
+# Default timeout for waiting for CI to complete (5 minutes)
+DEFAULT_CI_TIMEOUT_SECONDS = 300
+# Poll interval when waiting for CI
+CI_POLL_INTERVAL_SECONDS = 15
+
+
+class CIStatus(Enum):
+    """Status of CI checks on a PR."""
+
+    PASSED = "passed"  # All checks passed
+    FAILED = "failed"  # At least one check failed
+    PENDING = "pending"  # Checks are still running
+    UNKNOWN = "unknown"  # Could not determine status (e.g., API error)
+
+
+@dataclass
+class CIResult:
+    """Result of checking CI status."""
+
+    status: CIStatus
+    message: str
+    checks_total: int = 0
+    checks_passed: int = 0
+    checks_failed: int = 0
+    checks_pending: int = 0
+
+    @property
+    def is_complete(self) -> bool:
+        """True if CI is no longer running (passed, failed, or unknown)."""
+        return self.status in (CIStatus.PASSED, CIStatus.FAILED, CIStatus.UNKNOWN)
+
+    def to_dict(self) -> dict[str, Any]:
+        """Convert to dictionary for logging/debugging."""
+        return {
+            "status": self.status.value,
+            "message": self.message,
+            "checks_total": self.checks_total,
+            "checks_passed": self.checks_passed,
+            "checks_failed": self.checks_failed,
+            "checks_pending": self.checks_pending,
+        }
 
 
 class DoctorFailureMode(Enum):
@@ -145,6 +189,43 @@ class DoctorPhase:
                 phase_name="doctor",
                 data=diagnostics.to_dict(),
             )
+
+        # If doctor made commits, wait for CI to complete before proceeding
+        # This prevents validation from failing due to CI still running
+        if diagnostics.made_progress:
+            ci_result = self._wait_for_ci(ctx)
+            diagnostics_dict = diagnostics.to_dict()
+            diagnostics_dict["ci_status"] = ci_result.to_dict()
+
+            if ci_result.status == CIStatus.FAILED:
+                # CI failed after doctor's changes - this is a real failure
+                log_warning(f"[doctor] CI failed after doctor commits: {ci_result.message}")
+                diagnostics.failure_mode = DoctorFailureMode.INSUFFICIENT_CHANGES
+                return PhaseResult(
+                    status=PhaseStatus.FAILED,
+                    message=f"CI failed after doctor fixes: {ci_result.message}",
+                    phase_name="doctor",
+                    data=diagnostics_dict,
+                )
+
+            if ci_result.status == CIStatus.PENDING:
+                # CI timed out - report as warning but continue to validation
+                # The Judge phase will also check CI status
+                log_warning(
+                    f"[doctor] CI still pending after timeout, proceeding with validation"
+                )
+                ctx.report_milestone(
+                    "heartbeat",
+                    action="CI still pending after timeout, proceeding with validation",
+                )
+
+            # Check for shutdown during CI wait
+            if ci_result.status == CIStatus.UNKNOWN and "Shutdown" in ci_result.message:
+                return PhaseResult(
+                    status=PhaseStatus.SHUTDOWN,
+                    message="shutdown signal received while waiting for CI",
+                    phase_name="doctor",
+                )
 
         # Validate phase - doctor must re-request review
         if not self.validate(ctx):
@@ -371,6 +452,199 @@ class DoctorPhase:
                 return True
 
         return False
+
+    def _get_ci_status(self, ctx: ShepherdContext) -> CIResult:
+        """Get the current CI status for the PR.
+
+        Uses ``gh pr view`` to inspect the status check rollup and determine
+        if checks have passed, failed, or are still pending.
+
+        Args:
+            ctx: Shepherd context with pr_number set
+
+        Returns:
+            CIResult with current status and check counts
+        """
+        if ctx.pr_number is None:
+            return CIResult(
+                status=CIStatus.UNKNOWN,
+                message="No PR number available",
+            )
+
+        result = subprocess.run(
+            [
+                "gh",
+                "pr",
+                "view",
+                str(ctx.pr_number),
+                "--json",
+                "statusCheckRollup",
+            ],
+            cwd=ctx.repo_root,
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+
+        if result.returncode != 0:
+            return CIResult(
+                status=CIStatus.UNKNOWN,
+                message=f"Failed to get PR status: {result.stderr.strip()}",
+            )
+
+        try:
+            import json
+
+            data = json.loads(result.stdout)
+        except (json.JSONDecodeError, ValueError) as e:
+            return CIResult(
+                status=CIStatus.UNKNOWN,
+                message=f"Failed to parse PR status: {e}",
+            )
+
+        checks = data.get("statusCheckRollup", [])
+        if not checks:
+            # No checks configured - treat as passed
+            return CIResult(
+                status=CIStatus.PASSED,
+                message="No CI checks configured",
+                checks_total=0,
+            )
+
+        # Count check states
+        passed = 0
+        failed = 0
+        pending = 0
+
+        for check in checks:
+            conclusion = check.get("conclusion", "")
+            status = check.get("status", "")
+
+            if conclusion in ("SUCCESS", "NEUTRAL", "SKIPPED"):
+                passed += 1
+            elif conclusion in ("FAILURE", "CANCELLED", "TIMED_OUT", "ACTION_REQUIRED"):
+                failed += 1
+            elif status in ("IN_PROGRESS", "QUEUED", "PENDING", "WAITING"):
+                pending += 1
+            elif conclusion == "":
+                # Empty conclusion with no status usually means pending
+                pending += 1
+            else:
+                # Unknown state - treat as failed to be safe
+                failed += 1
+
+        total = passed + failed + pending
+
+        if failed > 0:
+            return CIResult(
+                status=CIStatus.FAILED,
+                message=f"CI failed: {failed}/{total} checks failed",
+                checks_total=total,
+                checks_passed=passed,
+                checks_failed=failed,
+                checks_pending=pending,
+            )
+
+        if pending > 0:
+            return CIResult(
+                status=CIStatus.PENDING,
+                message=f"CI pending: {pending}/{total} checks still running",
+                checks_total=total,
+                checks_passed=passed,
+                checks_failed=failed,
+                checks_pending=pending,
+            )
+
+        return CIResult(
+            status=CIStatus.PASSED,
+            message=f"CI passed: {passed}/{total} checks succeeded",
+            checks_total=total,
+            checks_passed=passed,
+            checks_failed=failed,
+            checks_pending=pending,
+        )
+
+    def _wait_for_ci(
+        self,
+        ctx: ShepherdContext,
+        timeout_seconds: int = DEFAULT_CI_TIMEOUT_SECONDS,
+    ) -> CIResult:
+        """Wait for CI checks to complete on the PR.
+
+        Polls the PR status at regular intervals until CI completes or times out.
+        Reports progress via heartbeat milestones so the operator can see status.
+
+        Args:
+            ctx: Shepherd context with pr_number set
+            timeout_seconds: Maximum time to wait for CI (default 5 minutes)
+
+        Returns:
+            CIResult with final status:
+            - PASSED: All checks passed
+            - FAILED: At least one check failed
+            - PENDING: Timeout reached while checks still running
+            - UNKNOWN: Error fetching status
+        """
+        start_time = time.time()
+        last_status_message = ""
+
+        log_info(f"[doctor] Waiting for CI checks on PR #{ctx.pr_number}")
+        ctx.report_milestone(
+            "heartbeat",
+            action=f"waiting for CI checks on PR #{ctx.pr_number}",
+        )
+
+        while True:
+            ci_result = self._get_ci_status(ctx)
+
+            # Log status changes
+            if ci_result.message != last_status_message:
+                log_info(f"[doctor] CI status: {ci_result.message}")
+                last_status_message = ci_result.message
+
+            # If CI is complete (passed, failed, or error), return immediately
+            if ci_result.is_complete:
+                log_info(f"[doctor] CI complete: {ci_result.status.value}")
+                return ci_result
+
+            # Check timeout
+            elapsed = time.time() - start_time
+            if elapsed >= timeout_seconds:
+                log_warning(
+                    f"[doctor] CI timeout after {int(elapsed)}s - "
+                    f"{ci_result.checks_pending} checks still pending"
+                )
+                ctx.report_milestone(
+                    "heartbeat",
+                    action=f"CI timeout: {ci_result.checks_pending} checks still pending",
+                )
+                # Return pending status to indicate timeout, not failure
+                return CIResult(
+                    status=CIStatus.PENDING,
+                    message=f"CI timeout after {int(elapsed)}s: {ci_result.checks_pending} checks still running",
+                    checks_total=ci_result.checks_total,
+                    checks_passed=ci_result.checks_passed,
+                    checks_failed=ci_result.checks_failed,
+                    checks_pending=ci_result.checks_pending,
+                )
+
+            # Check for shutdown signal
+            if ctx.check_shutdown():
+                log_info("[doctor] Shutdown signal detected while waiting for CI")
+                return CIResult(
+                    status=CIStatus.UNKNOWN,
+                    message="Shutdown signal received",
+                )
+
+            # Report periodic progress
+            if int(elapsed) % 60 == 0 and elapsed > 0:
+                ctx.report_milestone(
+                    "heartbeat",
+                    action=f"waiting for CI: {ci_result.checks_pending} checks pending ({int(elapsed)}s elapsed)",
+                )
+
+            # Wait before next poll
+            time.sleep(CI_POLL_INTERVAL_SECONDS)
 
     def run_test_fix(
         self, ctx: ShepherdContext, test_failure_data: dict[str, Any]

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -4894,13 +4894,18 @@ class TestDoctorPhaseExitCode5:
         self, mock_context: MagicMock
     ) -> None:
         """Exit code 0 should validate phase and return SUCCESS."""
-        from loom_tools.shepherd.phases.doctor import DoctorDiagnostics
+        from loom_tools.shepherd.phases.doctor import (
+            CIResult,
+            CIStatus,
+            DoctorDiagnostics,
+        )
 
         doctor = DoctorPhase()
         mock_context.pr_number = 123
         mock_context.check_shutdown.return_value = False
 
         diagnostics = DoctorDiagnostics(commits_made=1)
+        ci_result = CIResult(status=CIStatus.PASSED, message="CI passed")
 
         with (
             patch(
@@ -4909,6 +4914,7 @@ class TestDoctorPhaseExitCode5:
             patch.object(doctor, "validate") as mock_validate,
             patch.object(doctor, "_get_commit_count", return_value=0),
             patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result),
         ):
             mock_run.return_value = 0
             mock_validate.return_value = True
@@ -6524,6 +6530,8 @@ class TestDoctorFailureModes:
     ) -> None:
         """Doctor with exit 0 but validation failure should be validation_failed."""
         from loom_tools.shepherd.phases.doctor import (
+            CIResult,
+            CIStatus,
             DoctorDiagnostics,
             DoctorFailureMode,
         )
@@ -6534,6 +6542,7 @@ class TestDoctorFailureModes:
 
         # Create diagnostics that will be mutated
         diagnostics = DoctorDiagnostics(commits_made=1, pr_labels=["loom:changes-requested"])
+        ci_result = CIResult(status=CIStatus.PASSED, message="CI passed")
 
         with (
             patch(
@@ -6541,6 +6550,7 @@ class TestDoctorFailureModes:
             ) as mock_run,
             patch.object(doctor, "_get_commit_count", return_value=0),
             patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result),
             patch.object(doctor, "validate", return_value=False),
             patch.object(doctor, "_attempt_label_recovery", return_value=False),
         ):
@@ -6554,7 +6564,7 @@ class TestDoctorFailureModes:
         self, mock_context: MagicMock
     ) -> None:
         """When validation fails but doctor made progress, should attempt recovery."""
-        from loom_tools.shepherd.phases.doctor import DoctorDiagnostics
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorDiagnostics
 
         doctor = DoctorPhase()
         mock_context.pr_number = 123
@@ -6563,6 +6573,7 @@ class TestDoctorFailureModes:
         diagnostics = DoctorDiagnostics(
             commits_made=1, pr_labels=["loom:changes-requested"]
         )
+        ci_result = CIResult(status=CIStatus.PASSED, message="CI passed")
 
         with (
             patch(
@@ -6570,6 +6581,7 @@ class TestDoctorFailureModes:
             ) as mock_run,
             patch.object(doctor, "_get_commit_count", return_value=0),
             patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result),
             patch.object(doctor, "validate", return_value=False),
             patch.object(doctor, "_attempt_label_recovery") as mock_recovery,
         ):
@@ -6829,3 +6841,467 @@ class TestDoctorTestFixWithDiagnostics:
 
         assert result.status == PhaseStatus.SUCCESS
         assert result.data.get("commits_made") == 2
+
+
+class TestDoctorCIWaiting:
+    """Test Doctor phase CI waiting functionality (issue #2082)."""
+
+    def test_get_ci_status_returns_passed_when_all_checks_succeed(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return PASSED when all checks succeed."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+
+        mock_result = MagicMock()
+        mock_result.returncode = 0
+        mock_result.stdout = json.dumps({
+            "statusCheckRollup": [
+                {"conclusion": "SUCCESS", "status": "COMPLETED"},
+                {"conclusion": "SUCCESS", "status": "COMPLETED"},
+            ]
+        })
+
+        with patch("subprocess.run", return_value=mock_result):
+            ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.PASSED
+        assert ci_result.checks_passed == 2
+        assert ci_result.checks_failed == 0
+        assert ci_result.checks_pending == 0
+
+    def test_get_ci_status_returns_failed_when_check_fails(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return FAILED when any check fails."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+
+        mock_result = MagicMock()
+        mock_result.returncode = 0
+        mock_result.stdout = json.dumps({
+            "statusCheckRollup": [
+                {"conclusion": "SUCCESS", "status": "COMPLETED"},
+                {"conclusion": "FAILURE", "status": "COMPLETED"},
+            ]
+        })
+
+        with patch("subprocess.run", return_value=mock_result):
+            ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.FAILED
+        assert ci_result.checks_passed == 1
+        assert ci_result.checks_failed == 1
+
+    def test_get_ci_status_returns_pending_when_checks_running(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return PENDING when checks are in progress."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+
+        mock_result = MagicMock()
+        mock_result.returncode = 0
+        mock_result.stdout = json.dumps({
+            "statusCheckRollup": [
+                {"conclusion": "SUCCESS", "status": "COMPLETED"},
+                {"conclusion": "", "status": "IN_PROGRESS"},
+            ]
+        })
+
+        with patch("subprocess.run", return_value=mock_result):
+            ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.PENDING
+        assert ci_result.checks_passed == 1
+        assert ci_result.checks_pending == 1
+
+    def test_get_ci_status_returns_passed_when_no_checks_configured(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return PASSED when no checks are configured."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+
+        mock_result = MagicMock()
+        mock_result.returncode = 0
+        mock_result.stdout = json.dumps({"statusCheckRollup": []})
+
+        with patch("subprocess.run", return_value=mock_result):
+            ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.PASSED
+        assert "No CI checks configured" in ci_result.message
+
+    def test_get_ci_status_returns_unknown_on_api_error(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return UNKNOWN when gh command fails."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+
+        mock_result = MagicMock()
+        mock_result.returncode = 1
+        mock_result.stderr = "API error"
+
+        with patch("subprocess.run", return_value=mock_result):
+            ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.UNKNOWN
+        assert "Failed to get PR status" in ci_result.message
+
+    def test_get_ci_status_returns_unknown_when_no_pr_number(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_get_ci_status should return UNKNOWN when no PR number."""
+        from loom_tools.shepherd.phases.doctor import CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = None
+
+        ci_result = doctor._get_ci_status(mock_context)
+
+        assert ci_result.status == CIStatus.UNKNOWN
+        assert "No PR number" in ci_result.message
+
+    def test_wait_for_ci_returns_immediately_when_passed(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_wait_for_ci should return immediately when CI passes."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        passed_result = CIResult(
+            status=CIStatus.PASSED,
+            message="CI passed",
+            checks_total=2,
+            checks_passed=2,
+        )
+
+        with patch.object(doctor, "_get_ci_status", return_value=passed_result):
+            ci_result = doctor._wait_for_ci(mock_context, timeout_seconds=60)
+
+        assert ci_result.status == CIStatus.PASSED
+
+    def test_wait_for_ci_returns_immediately_when_failed(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_wait_for_ci should return immediately when CI fails."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        failed_result = CIResult(
+            status=CIStatus.FAILED,
+            message="CI failed",
+            checks_total=2,
+            checks_passed=1,
+            checks_failed=1,
+        )
+
+        with patch.object(doctor, "_get_ci_status", return_value=failed_result):
+            ci_result = doctor._wait_for_ci(mock_context, timeout_seconds=60)
+
+        assert ci_result.status == CIStatus.FAILED
+
+    def test_wait_for_ci_polls_until_complete(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_wait_for_ci should poll until CI completes."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        # First two calls return pending, third returns passed
+        pending_result = CIResult(
+            status=CIStatus.PENDING,
+            message="CI pending",
+            checks_total=2,
+            checks_passed=1,
+            checks_pending=1,
+        )
+        passed_result = CIResult(
+            status=CIStatus.PASSED,
+            message="CI passed",
+            checks_total=2,
+            checks_passed=2,
+        )
+
+        call_count = [0]
+
+        def mock_get_ci_status(ctx: Any) -> CIResult:
+            call_count[0] += 1
+            if call_count[0] < 3:
+                return pending_result
+            return passed_result
+
+        with (
+            patch.object(doctor, "_get_ci_status", side_effect=mock_get_ci_status),
+            patch("loom_tools.shepherd.phases.doctor.time.sleep"),
+        ):
+            ci_result = doctor._wait_for_ci(mock_context, timeout_seconds=300)
+
+        assert ci_result.status == CIStatus.PASSED
+        assert call_count[0] == 3
+
+    def test_wait_for_ci_returns_pending_on_timeout(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_wait_for_ci should return PENDING status on timeout."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        pending_result = CIResult(
+            status=CIStatus.PENDING,
+            message="CI pending",
+            checks_total=2,
+            checks_passed=1,
+            checks_pending=1,
+        )
+
+        with (
+            patch.object(doctor, "_get_ci_status", return_value=pending_result),
+            patch("loom_tools.shepherd.phases.doctor.time.sleep"),
+            patch("loom_tools.shepherd.phases.doctor.time.time", side_effect=[0, 0, 400]),
+        ):
+            ci_result = doctor._wait_for_ci(mock_context, timeout_seconds=300)
+
+        assert ci_result.status == CIStatus.PENDING
+        assert "timeout" in ci_result.message.lower()
+
+    def test_wait_for_ci_handles_shutdown_signal(
+        self, mock_context: MagicMock
+    ) -> None:
+        """_wait_for_ci should return UNKNOWN on shutdown signal."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.side_effect = [False, True]
+
+        pending_result = CIResult(
+            status=CIStatus.PENDING,
+            message="CI pending",
+        )
+
+        with (
+            patch.object(doctor, "_get_ci_status", return_value=pending_result),
+            patch("loom_tools.shepherd.phases.doctor.time.sleep"),
+        ):
+            ci_result = doctor._wait_for_ci(mock_context, timeout_seconds=300)
+
+        assert ci_result.status == CIStatus.UNKNOWN
+        assert "Shutdown" in ci_result.message
+
+    def test_run_waits_for_ci_when_doctor_makes_commits(
+        self, mock_context: MagicMock
+    ) -> None:
+        """run() should wait for CI when doctor makes commits."""
+        from loom_tools.shepherd.phases.doctor import (
+            CIResult,
+            CIStatus,
+            DoctorDiagnostics,
+            DoctorPhase,
+        )
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        diagnostics = DoctorDiagnostics(
+            commits_made=1,
+            pr_labels=["loom:review-requested"],
+        )
+
+        ci_result = CIResult(
+            status=CIStatus.PASSED,
+            message="CI passed",
+            checks_total=2,
+            checks_passed=2,
+        )
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.doctor.run_phase_with_retry"
+            ) as mock_run,
+            patch.object(doctor, "_get_commit_count", return_value=0),
+            patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result) as mock_wait,
+            patch.object(doctor, "validate", return_value=True),
+        ):
+            mock_run.return_value = 0
+            result = doctor.run(mock_context)
+
+        # CI wait should have been called
+        mock_wait.assert_called_once_with(mock_context)
+        assert result.status == PhaseStatus.SUCCESS
+
+    def test_run_returns_failed_when_ci_fails_after_commits(
+        self, mock_context: MagicMock
+    ) -> None:
+        """run() should return FAILED when CI fails after doctor commits."""
+        from loom_tools.shepherd.phases.doctor import (
+            CIResult,
+            CIStatus,
+            DoctorDiagnostics,
+            DoctorPhase,
+        )
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        diagnostics = DoctorDiagnostics(commits_made=1)
+
+        ci_result = CIResult(
+            status=CIStatus.FAILED,
+            message="CI failed: 1/2 checks failed",
+            checks_total=2,
+            checks_passed=1,
+            checks_failed=1,
+        )
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.doctor.run_phase_with_retry"
+            ) as mock_run,
+            patch.object(doctor, "_get_commit_count", return_value=0),
+            patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result),
+        ):
+            mock_run.return_value = 0
+            result = doctor.run(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert "CI failed" in result.message
+        assert result.data.get("ci_status", {}).get("status") == "failed"
+
+    def test_run_continues_to_validation_when_ci_times_out(
+        self, mock_context: MagicMock
+    ) -> None:
+        """run() should continue to validation when CI times out (pending)."""
+        from loom_tools.shepherd.phases.doctor import (
+            CIResult,
+            CIStatus,
+            DoctorDiagnostics,
+            DoctorPhase,
+        )
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        diagnostics = DoctorDiagnostics(
+            commits_made=1,
+            pr_labels=["loom:review-requested"],
+        )
+
+        ci_result = CIResult(
+            status=CIStatus.PENDING,
+            message="CI timeout: 1 checks still running",
+            checks_total=2,
+            checks_passed=1,
+            checks_pending=1,
+        )
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.doctor.run_phase_with_retry"
+            ) as mock_run,
+            patch.object(doctor, "_get_commit_count", return_value=0),
+            patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci", return_value=ci_result),
+            patch.object(doctor, "validate", return_value=True) as mock_validate,
+        ):
+            mock_run.return_value = 0
+            result = doctor.run(mock_context)
+
+        # Should still proceed to validation
+        mock_validate.assert_called_once()
+        assert result.status == PhaseStatus.SUCCESS
+
+    def test_run_does_not_wait_for_ci_when_no_commits(
+        self, mock_context: MagicMock
+    ) -> None:
+        """run() should not wait for CI when doctor makes no commits."""
+        from loom_tools.shepherd.phases.doctor import DoctorDiagnostics, DoctorPhase
+
+        doctor = DoctorPhase()
+        mock_context.pr_number = 123
+        mock_context.check_shutdown.return_value = False
+
+        diagnostics = DoctorDiagnostics(
+            commits_made=0,
+            pr_labels=["loom:review-requested"],
+        )
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.doctor.run_phase_with_retry"
+            ) as mock_run,
+            patch.object(doctor, "_get_commit_count", return_value=0),
+            patch.object(doctor, "_diagnose_doctor_outcome", return_value=diagnostics),
+            patch.object(doctor, "_wait_for_ci") as mock_wait,
+            patch.object(doctor, "validate", return_value=True),
+        ):
+            mock_run.return_value = 0
+            doctor.run(mock_context)
+
+        # CI wait should NOT have been called since no commits made
+        mock_wait.assert_not_called()
+
+    def test_ci_result_is_complete_property(self) -> None:
+        """CIResult.is_complete should return True for terminal states."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus
+
+        passed = CIResult(status=CIStatus.PASSED, message="")
+        failed = CIResult(status=CIStatus.FAILED, message="")
+        pending = CIResult(status=CIStatus.PENDING, message="")
+        unknown = CIResult(status=CIStatus.UNKNOWN, message="")
+
+        assert passed.is_complete is True
+        assert failed.is_complete is True
+        assert pending.is_complete is False
+        assert unknown.is_complete is True
+
+    def test_ci_result_to_dict(self) -> None:
+        """CIResult.to_dict should include all fields."""
+        from loom_tools.shepherd.phases.doctor import CIResult, CIStatus
+
+        ci_result = CIResult(
+            status=CIStatus.PASSED,
+            message="CI passed",
+            checks_total=3,
+            checks_passed=2,
+            checks_failed=0,
+            checks_pending=1,
+        )
+
+        result = ci_result.to_dict()
+
+        assert result["status"] == "passed"
+        assert result["message"] == "CI passed"
+        assert result["checks_total"] == 3
+        assert result["checks_passed"] == 2
+        assert result["checks_failed"] == 0
+        assert result["checks_pending"] == 1

